crates/polars-plan/src/logical_plan/builder.rs

```diff
@@ -616,6 +616,27 @@ impl LogicalPlanBuilder {
         } else {
             predicate
         };
+
+        // Check predicates refer to valid column names here, as this is not
+        // checked by predicate pushdown and may otherwise lead to incorrect
+        // optimizations. For example:
+        //
+        // (unoptimized)
+        // FILTER [(col("x")) == (1)] FROM
+        //   SELECT [col("x").alias("y")] FROM
+        //     DF ["x"]; PROJECT 1/1 COLUMNS; SELECTION: "None"
+        //
+        // (optimized)
+        // SELECT [col("x").alias("y")] FROM
+        //   DF ["x"]; PROJECT 1/1 COLUMNS; SELECTION: "[(col(\"x\")) == (1)]"
+        //                                             ^^^
+        // 'x' is incorrectly pushed down even though it didn't exist after selection
+        try_delayed!(
+            expressions_to_schema(&[predicate.clone()], &schema, Context::Default),
```

Comment:
We can do this check without materializing a new schema. That's a bit wasteful.

Can you rewrite so that we only check if the predicates refer to valid columns without needlessly allocating?